### PR TITLE
Image Customizer: Fix ISO gen with SELinux disabled base image.

### PR DESCRIPTION
**Pull request by cwize1**
Currently, the ISO gen disables SELinux in the image because there is an issue causing the ISO not to boot if SELinux is enabled. However, the logic to disable SELinux fails if SELinux is already disabled. This change fixes this problem.

This change also combines the two different implementations for changing the SELinux mode (in the kernel CLI args). It also makes the logic more robust to the ordering of the kernel CLI args.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Manually ran image customizer tool.

### CODE REVIEW

File: toolkit/tools/pkg/imagecustomizerlib/customizeboot.go [hunk 1]
```diff
@@ -138,10 +127,10 @@ func setLinuxPath(inputGrubCfgContent string, linuxPath string) (outputGrubCfgCo
 func setInitrdPath(inputGrubCfgContent string, initrdPath string) (outputGrubCfgContent string, oldInitrdPath string, err error) {
 	match := initPathRegex.FindStringSubmatchIndex(inputGrubCfgContent)
 	if match == nil {
-		return "", "", fmt.Errorf("failed to find initrd command line.")
+		return "", "", fmt.Errorf("failed to find initrd command line")
 	}
 	if len(match) <= initPathRegexPathIndexEnd {
-		return "", "", fmt.Errorf("failed to find initrd command line path.")
+		return "", "", fmt.Errorf("failed to find initrd command line path")
```
Comment (romoh) [inline toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:133]: nit - use a different error so you can differentiate between this failure and the previous one

File: toolkit/tools/pkg/imagecustomizerlib/customizeboot.go [hunk 2]
```diff
@@ -170,40 +159,78 @@ func appendKernelCommandLineArguments(inputGrubCfgContent string, extraCommandLi
...
+	}
 
-	oldValue = inputGrubCfgContent[start:end]
-	outputGrubCfgContent = inputGrubCfgContent[:start] + value + inputGrubCfgContent[end:]
+	outputGrubCfgContent = fmt.Sprintf("%s%s=%s%s", inputGrubCfgContent[:start], name, value, inputGrubCfgContent[end:])
 
 	return outputGrubCfgContent, oldValue, nil
 }
 
+func tryRemoveKernelCommandLineArgumentValue(inputGrubCfgContent string, name string) (string, int, error) {
```
Comment (romoh) [inline toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:217]: why is this tryRemove..()? what expectations should caller of this have?

File: toolkit/tools/pkg/imagecustomizerlib/customizeboot.go (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:133]: They are different. By 1 word.

File: toolkit/tools/pkg/imagecustomizerlib/customizeboot.go (hunk 2 above)
Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:229]: Add comment.

File: toolkit/tools/pkg/imagecustomizerlib/customizeboot.go [hunk 3]
```diff
@@ -170,40 +159,90 @@ func appendKernelCommandLineArguments(inputGrubCfgContent string, extraCommandLi
...
 }
 
-func replaceKernelCommandLineArgumentValue(inputGrubCfgContent string, name string, value string) (outputGrubCfgContent string, oldValue string, err error) {
+// Tries to find the specified kernel CLI arg. Does not fail if the arg is not found.
+//
+// Returns:
+// - startIndex: The string index where the arg was found. Or -1 if not found.
+// - endIndex: The string index of the end of the arg. Or -1 if not found.
+// - value: The value of the arg. Or "" if not found.
+func findKernelCommandLineArgumentArg(grub2Config string, name string) (int, int, string, error) {
```
Comment (dmcilvaney) [inline toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:168]: edge case, but what happens if the same key is used twice? Some cmdline args support that additively (crash kernel size for example), while others just take the last in the list.

Comment (cwize1) [reply toolkit/tools/pkg/imagecustomizerlib/customizeboot.go:168]: This code actually has a lot of problems, such as being unable to handle escape sequences and single/double quoted strings. So, only really aiming for good enough at the moment.

**Review by romoh — APPROVED** [review]

**Review by amritakohli — APPROVED** [review]

**Pull request merged**